Gthorbranch/jthor/initial flask code

GerThor opened the pull request:
Added some work onto "jthor/initial" flask code branch for a draft pull request.

GerThor committed:
Worked on API Calls Coding

Got an API Call to work for City Names, which will obtain a json object containing longitude and latitude data which will be used for another API Call to obtain more valuable weather data. The API Call to obtain weather data uses longitude and latitude fields instead of city/state/country names, so a previous API call must be used to convert city/state/country names into geo coordinates.

JackThor commented on main.py:
```diff
@@ -0,0 +1,84 @@
+
+
+
+    response = requests.get(f"http://api.openweathermap.org/geo/1.0/direct?q=[{city}]&limit=5&appid=bd3844093fc57703b2cc41751e23f944") #Currently Using Oroville as static API Call, change to [city] later
```
Refactor the code. There should at least be 3 object. One is the service that makes the API call, one is the object that contains the information we need to a pass, and one for the response we get back. This will help make the code more testable, clean, easier to serialize and maintain.  The service object should have error handling in situation where the api call fails. Also look into logging library so we can write the errors to a log.

JackThor reviewed (changes requested):
Make the code more modular it will help with keeping the logic separate and more readable.

GerThor committed:
Added More API Calling Code

Able to obtain specific US city and state longitude and latitude coordinates. However, am failing to make an API call to obtain current weather data. Failing with Error 401, representing unauthorized permissions error

JackThor commented on main.py:
```diff
@@ -0,0 +1,111 @@
+            print("Dallas,TX,US Latitude:", latitude)
+
+            #Code is failing, I believe it is failing on the request line below
+            weather_response = requests.get(f"https://api.openweathermap.org/data/3.0/onecall?lat=32.77&lon=-96.79&exclude=minutely,hourly,daily,alerts&appid=bd3844093fc57703b2cc41751e23f944")
```
you have to use the free one, the OneCall Api cost money. Use this one, https://api.openweathermap.org/data/2.5/weather

GerThor committed:
Added more work to main.py

Added switch cases for user input of states into state codes. Added a bit of formatting for user input of cities.

GerThor committed:
Modified code to use classes and objects

Needs a bit more cleaning up, and some error handling, will make another function in Weather() Class to run the required functions so in main.py it will just be a one liner code to get weather data

JackThor commented on main.py:
```diff
@@ -0,0 +1,180 @@
+
+
+
+# def state_code_convert(state):
```
Remvoe repeated code

JackThor commented on main.py:
```diff
@@ -0,0 +1,180 @@
+#     return state_code
+
+
+# def get_coordinates(city_input, state_code, country_code, api_key):
```
remove repeated code

JackThor reviewed (changes requested):
Clean up code and remove the printf that is for debugging, like the ones that show spit back out the data. Keep the ones that track success and failure.

GerThor committed:
Cleaned up old comemnts / added new comments

Cleaned up old commented code, and added TO DO's in comments.
